mobile/library/common/http/client.h

```diff
@@ -74,11 +76,12 @@ class Client : public Logger::Loggable<Logger::Id::http> {
   /**
    * Send headers over an open HTTP stream. This method can be invoked once and needs to be called
    * before send_data.
-   * @param stream, the stream to send headers over.
-   * @param headers, the headers to send.
-   * @param end_stream, indicates whether to close the stream locally after sending this frame.
+   * @param stream the stream to send headers over.
+   * @param header_func the function that takes the header reference to be modified.
```

Comment:
can we enhance this comment?  sendHeaders with headers is pretty clear.  sendHeaders with headers function it's less clear what the function is for.  Again maybe we can just use the base envoy header class, and change ownership?

Reply:
I updated the comment.

The reason why it takes a function is because we want `client.cc` to be responsible for creating the `RequestHeaderMap`. In this case it is the `RequestHeaderMap` with the preserve case header formatter. If this function simply took `RequestHeaderMap` parameter, there would be no way to control that behavior and then each language binding would have to construct the `RequestHeaderMap` with the correct formatter. In the follow-up PR, I plan on updating `Stream::sendHeaders` to also take the `header_func`. This will also make the C++ API consistent with the rest of the language APIs.